### PR TITLE
Featurebranch version4

### LINKED ISSUE #5: Update plugin to use Leapwork API v4
Current version is targeting to use API v3 which must be deprecated. We need to update plugin to use API v4

### PR DESCRIPTION
**Pull request by BirSikanderMahajan**
Updated the following:
- Version 4 from V3
- CSP Headers
- Folder Structure
- Plugin Versioning
- Reporting

### CODE REVIEW

File: build/pom.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      </dependency>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-common</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
+      </dependency>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-server</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline build/pom.xml]: 4.0.1 please

File: build/pom.xml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  <dependencies>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-agent</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
+      </dependency>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-common</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline build/pom.xml]: 4.0.1 please

File: build/pom.xml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      <maven.build.timestamp.format>yyyyMMddHHmmss</maven.build.timestamp.format>
+      <teamcity.leapwork-integration.plugin.version>snapshot-${maven.build.timestamp}</teamcity.leapwork-integration.plugin.version>
+      <teamcity.leapwork-integration.plugin.vendorName>Plugin vendor name</teamcity.leapwork-integration.plugin.vendorName>
+      <teamcity.leapwork-integration.plugin.vendorUrl>Plugin vendor URL</teamcity.leapwork-integration.plugin.vendorUrl>
+  </properties>
+  <dependencies>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-agent</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline build/pom.xml]: 4.0.1

File: README.md [hunk 4]
```diff
@@ -1 +1,41 @@
...
+ 
+# Installing
+- Use maven.
+- Command: mvn package 
+- Or simply install zip-file from the "target" folder: Copy the zip plugin package into the {TeamCity Data Directory}/plugins directory (Default path: C:\ProgramData\JetBrains\TeamCity\plugins). 
+- If you have an earlier version of the plugin in the directory, remove it.
+- Alternatively, use the Administration -> Plugins List -> Upload plugin zip -> Choose File -> Choose that zip-file -> Press Upload plugin zip
+
+# Update 3.1.1
+- For LEAPWORK version 2018.2.301
```
Comment (RomanSyrychLeapwork) [inline README.md]: Readme should be updated

File: README.md [hunk 5]
```diff
@@ -1 +1,41 @@
...
+# Installing
+- Use maven.
+- Command: mvn package 
+- Or simply install zip-file from the "target" folder: Copy the zip plugin package into the {TeamCity Data Directory}/plugins directory (Default path: C:\ProgramData\JetBrains\TeamCity\plugins). 
+- If you have an earlier version of the plugin in the directory, remove it.
+- Alternatively, use the Administration -> Plugins List -> Upload plugin zip -> Choose File -> Choose that zip-file -> Press Upload plugin zip
+
+# Update 4.0.1
+- For Leapwork version 2021.1
+- Fixed bug when schedules are executing in non-ordered way.
```
Comment (AntonLiakhovich) [inline README.md]: Replace all update section to
# Update 4.0.1
- Leapwork version 2021.1 is now supported
- Plugin is now using Leapwork API v4

File: build/pom.xml (hunk 3 above)
Comment (AntonLiakhovich) [inline build/pom.xml]: Check why this is not updated

**Review by AntonLiakhovich — CHANGES_REQUESTED** [review]
see my comments

**Review by AntonLiakhovich — APPROVED** [review]
Looks fine for me. @RomanSyrychLeapwork it still required your attention

File: README.md (hunk 4 above)
Comment (AntonLiakhovich) [reply README.md]: it's updated already, check latest commits

**Review by RomanSyrychLeapwork — APPROVED** [review]

**Pull request merged**